service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -99,64 +96,132 @@ public void setRestTemplate(RestTemplate restTemplate) {
+            return new ResponseEntity<>(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
+    public ResponseEntity<?> generate(final String msgProtocol, final String msgType, final Boolean failIfMultipleFound, final Boolean failIfNoneFound, final Boolean lookupInExternalERs, final int lookupLimit, final Boolean okToLeaveOutInvalidOptionalFields, JsonElement bodyJson) {
+
+        JsonArray results = new JsonArray();
+        JsonObject errorResponse = new JsonObject();
         try {
-            bodyJson = erLookup(bodyJson, failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit);
+
+            if (bodyJson.isJsonArray()) {
+
+                JsonArray jsonArray = bodyJson.getAsJsonArray();
+                for (JsonElement element : jsonArray) {
+                    results.add(processEvent(element.getAsJsonObject(), msgProtocol, msgType,
+                            failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit,
+                            okToLeaveOutInvalidOptionalFields));
+                }
+                return new ResponseEntity<>(results, HttpStatus.OK);
+
+            } else if (bodyJson.isJsonObject()) {
+                JsonObject jsonObject = bodyJson.getAsJsonObject();
+                JsonObject jsonObject1 = processEvent(jsonObject, msgProtocol, msgType, failIfMultipleFound, failIfNoneFound,
+                        lookupInExternalERs, lookupLimit, okToLeaveOutInvalidOptionalFields);
+                return new ResponseEntity<>(jsonObject1, HttpStatus.OK);
+            } else {
+                errorResponse.addProperty("Status code", HttpStatus.BAD_REQUEST.value());
+                errorResponse.addProperty("result", "fail");
+                errorResponse.addProperty("message", "Invalid JSON format");
+                return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+            }
+        } catch (Exception e) {
+            log.error("Unexpected exception caught", e);
+            errorResponse.addProperty("Status code", HttpStatus.BAD_REQUEST.value());
+            errorResponse.addProperty("Result", "Fail");
+            errorResponse.addProperty("Message", "Invalid JSON format");
+            return new ResponseEntity<>(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+
+        }
+    }
+
+    /**
+     * This helper method basically generate or process one event
+     * @return JsonObject generated event
+     */
+
+    public JsonObject processEvent(JsonObject jsonElement, String msgProtocol, String msgType, Boolean failIfMultipleFound,
+                                   Boolean okToLeaveOutInvalidOptionalFields,
+                                   Boolean failIfNoneFound, int lookupLimit, Boolean lookupInExternalERs) {
+        JsonObject eventResponse = new JsonObject();
+        JsonElement parsedResponse = null;
+        try {
+            JsonObject event = jsonElement.getAsJsonObject();
```

Comment:
Why 'getAsJsonObject()'? It's JsonObject, see method header.

Reply:
yeah correct